src/main/java/uk/gov/hmcts/ccd/domain/service/processor/GlobalSearchProcessorService.java

```diff
@@ -0,0 +1,202 @@
+
+        Map<String, JsonNode> clonedData = null;
+        if (data != null) {
+            clonedData = new HashMap<>(data);
+        }
+
+        if (searchCriteriaCaseField.isPresent()) {
+
+            SearchCriteria searchCriteria = populateSearchCriteria(clonedData, searchCriterias);
+            List<SearchParty> searchPartyList = populateSearchParties(clonedData, searchParties);
+
+            if (!searchPartyList.isEmpty()) {
+                searchCriteria.setSearchParties(searchPartyList);
+            }
+            if (!searchCriteria.isEmpty() && clonedData != null) {
+                clonedData.put(SEARCH_CRITERIA, JacksonUtils.convertValueJsonNode(searchCriteria));
+            }
+        }
+
+        return clonedData;
+    }
+
+    private SearchCriteria populateSearchCriteria(Map<String, JsonNode> data,
+                                       List<uk.gov.hmcts.ccd.domain.model.definition.SearchCriteria> searchCriterias) {
+        List<OtherCaseReference> otherCaseReferences = new ArrayList<>();
+        SearchCriteria returnValue = new SearchCriteria();
+
+        if (data != null) {
+            data.forEach((key, jsonNode) -> searchCriterias.forEach(currentSearchCriteria -> {
+
+                if (currentSearchCriteria.getOtherCaseReference().contains(".")
+                    && key.equals(currentSearchCriteria.getOtherCaseReference().split("\\.")[0])) {
+                    JsonNode nestedCaseFieldByPath =
+                        CaseFieldPathUtils.getNestedCaseFieldByPath(jsonNode,
+                            currentSearchCriteria.getOtherCaseReference().substring(
+                                currentSearchCriteria.getOtherCaseReference().indexOf(".") + 1));
+                    otherCaseReferences.add(OtherCaseReference.builder()
+                        .id(UUID.randomUUID().toString())
+                        .value(nestedCaseFieldByPath.textValue())
+                        .build());
+                } else if (key.equals(currentSearchCriteria.getOtherCaseReference())) {
+                    otherCaseReferences.add(OtherCaseReference.builder()
+                        .id(UUID.randomUUID().toString())
+                        .value(jsonNode.textValue())
+                        .build());
+                }
+            }));
+
+            returnValue = SearchCriteria.builder().otherCaseReferences(otherCaseReferences).build();
+        }
+
+        return returnValue;
+    }
+
+    private List<SearchParty> populateSearchParties(Map<String, JsonNode> data,
+                                             List<uk.gov.hmcts.ccd.domain.model.definition.SearchParty> searchParties) {
+        List<SearchParty> searchPartyList = new ArrayList<>();
+
+        searchParties.forEach(searchParty -> {
+            SearchPartyValue valueToPopulate  = new SearchPartyValue();
+
+            data.forEach((key, jsonNode) -> populateSearchParty(searchParty, valueToPopulate, key, jsonNode));
```

Comment:
probably better to return `SearchPartyValue` from populateSearchParty(..) instead of changing passed value. we could even move `data.forEach` inside populateSearchParty(..) 💭

Reply:
fixed in b1524eb